### PR TITLE
Fix/425

### LINKED ISSUE #425: `get_horz_ngrams` works only when mention is_tabular
**Describe the bug**

I have a span mention, `mention`, that is visual but is not tabular.
I tried to get all horizontally aligned ngrams from the same sentence (ie `get_horz_ngrams(mention, from_sentence=False)`), but I got none because it is not tabular.

**To Reproduce**

```python
>>> mention = session.query(Mention).all()[0]
>>> print(mention.context.get_span())
Obama
>>> print(mention.context.sentence.text)
Obama was born in Honolulu, Hawaii.
>>> print(mention.context.sentence.is_visual())
True
>>> print(mention.context.sentence.is_tabular())
False
>>> from fonduer.utils.data_model_utils.visual import get_horz_ngrams
>>> print(list(get_horz_ngrams(mention, from_sentence=False)))
[]
```

**Expected behavior**

Since all the other 1-gram in the same sentence is horizontally aligned with the mention, I want to get all those.

```python
>>> print(list(get_horz_ngrams(mention, from_sentence=False)))
["was", "born", "in", "Honolulu", ",", "Hawaii", "."]
```

**Error Logs/Screenshots**
If applicable, add error logs or screenshots to help explain your problem.

**Environment (please complete the following information):**
 - OS: N/A
 - PostgreSQL Version: N/A
 - Poppler Utils Version: N/A
 - Fonduer Version: 0.8.2

**Additional context**

In addition to the above issue, I feel `from_sentence` is a bit confusing.

### PR DESCRIPTION
**Pull request by HiromuHota**
Fix #425.

### CODE REVIEW

**Comment by HiromuHota** [comment]
`from_sentence` looks erroneous. The description (document) and the behaviour (implementation) do not seem to match.
`from_sentence` was introduced by ff42e41e422ef7a02786dc1334cdb5e3862c66e2, where `from_sentence` was called `from_phrase`.

**Comment by HiromuHota** [comment]
I figured that out. `from_sentence` does what it is supposed to do. It was just confusing. I'll add comments and improve the documentation.

File: src/fonduer/utils/data_model_utils/visual.py [hunk 1]
```diff
@@ -236,35 +242,39 @@ def _get_direction_ngrams(
...
                         getattr(sentence, attrib), n_min=n_min, n_max=n_max, lower=lower
                     ):
                         yield ngram
             else:
                 for ts in ngrams_space.apply(sentence):
-                    if bbox_direction_aligned(
-                        bbox_from_span(ts), bbox_from_span(span)
-                    ) and not (
-                        sentence == span.sentence and ts.get_span() in span.get_span()
+                    if (  # True if visually aligned AND not from itsself.
```
Comment (lukehsiao) [inline src/fonduer/utils/data_model_utils/visual.py:274]:
```suggestion
                    if (  # True if visually aligned AND not from itself.
```

**Review by lukehsiao — CHANGES_REQUESTED** [review]

File: src/fonduer/utils/data_model_utils/visual.py (hunk 1 above)
Comment (HiromuHota) [reply src/fonduer/utils/data_model_utils/visual.py]: Thanks!

**Review by lukehsiao — APPROVED** [review]

**Comment by lukehsiao** [comment]
@HiromuHota could you squash these commits with a nice commit description, then we can merge?

**Comment by HiromuHota** [comment]
I added another unit test: `test_get_ngrams_that_match_in_string` (c55295d7439893ac80ef7a926817f1e701ddbba0).
Sorry for the last minute commit!
If this looks good to you, I'll squash into one single commit.

**Comment by lukehsiao** [comment]
Looks good to me!

**Comment by HiromuHota** [comment]
Squashed and force-pushed.
Please merge this at your convenience.

**Pull request merged**